common/src/main/java/org/apache/iceberg/common/DynConstructors.java

```diff
@@ -58,7 +58,6 @@ public C newInstanceChecked(Object... args) throws Exception {
         throw e;
       } catch (InvocationTargetException e) {
         Throwables.propagateIfInstanceOf(e.getCause(), Exception.class);
-        Throwables.propagateIfInstanceOf(e.getCause(), RuntimeException.class);
```

Comment:
I think this line is never reached as the previous line always catches even `RuntimeException`s, so it is safe to remove this.